IEX.h

```diff
@@ -85,310 +86,353 @@ namespace IEX {
-                cd.company_name    = jsonData["companyName"].asString();
-                cd.exchange        = jsonData["exchange"].asString();
-                cd.industry        = jsonData["industry"].asString();
-                cd.website         = jsonData["website"].asString();
-                cd.description     = jsonData["description"].asString();
-                cd.CEO             = jsonData["CEO"].asString();
-                cd.issue_type      = jsonData["issue_type"].asString();
-                cd.sector          = jsonData["sector"].asString();
-                // clang-format on
+            dividends_data.called_endpoint = url;
+            dividends_data.number_of_dividends = json_data.size();
+            std::vector<Dividend> dividends;
+            auto beginning = dividends.begin();
 
-                // Iterate over all tags and add to vector
-                auto tags = jsonData["tags"];
-                for (const auto& t : tags) {
-                    cd.tags.push_back(t.asString());
-                }
+            for (const auto& dividend_json : json_data) {
+                Dividend dividend;
+                dividend.ex_date = getString(dividend_json["exDate"]);
 
-                return cd;
-            }
-        };
-
-        struct PriceData {
-            std::string called_endpoint;
-            std::string stock_symbol;
-            double latest_price;
-        };
-
-        class Price {
-        public:
-            static std::size_t callback(const char* in,
-                                        std::size_t size,
-                                        std::size_t num,
-                                        std::string* out) {
-                const std::size_t totalBytes(size * num);
-                out->append(in, totalBytes);
-                return totalBytes;
-            }
+                dividend.payment_date = getString(dividend_json["paymentDate"]);
+
+                dividend.record_date = getString(dividend_json["recordDate"]);
 
-            static PriceData get(const std::string stock_symbol) {
-                std::string url = IEX_API_V1_ENDPOINT;
-                url += "/stock/" + std::string(stock_symbol) + "/price";
+                dividend.declared_date = getString(dividend_json["declaredDate"]);
 
-                Json::Value jsonData;
-                sendGetRequest(jsonData, url, callback);
+                dividend.amount = getDouble(dividend_json["amount"]);
 
-                PriceData pd;
-                pd.called_endpoint = url;
-                pd.stock_symbol = stock_symbol;
-                pd.latest_price = jsonData.asDouble();
+                dividend.flag = getString(dividend_json["flag"]);
 
-                return pd;
+                dividend.type = getString(dividend_json["type"]);
+
+                dividend.qualified = getString(dividend_json["qualified"]);
+
+                std::string indicated = getString(dividend_json["indicated"]);
```

Comment:
Why not just call `getDouble` here? Or add a `getFloat`?

Reply:
I had it set up differently earlier when developing. I changed it and forgot to make that change.

Reply:
Actually, on further review, I remembered that indicated is sometimes just an empty string and not a number, so this part is necessary.